### DISCUSSION TITLE
bao demo not running when smp number set to 2

### DISCUSSION THREAD

**Discussion by xioliu** [bao-project/bao-hypervisor#142] [General]
Hello, 

I am curious that when I tried to change the smp number to 2, bao-demo (baremetal) cannot run successfully, I have changed cpu_num to 2 in src\platform\qemu-aarch64-virt\virt_desc.c, may I know why? Thank you.

The original command is "qemu-system-aarch64 -nographic\
   -M virt,secure=on,virtualization=on,gic-version=3 \
   -cpu cortex-a53 **-smp 4** -m 4G\
   -bios $BAO_DEMOS_WRKDIR/imgs/$PLATFORM/flash.bin \
   -device loader,file="$BAO_DEMOS_WRKDIR_IMGS/bao.bin",addr=0x50000000,force-raw=on\
   -device virtio-net-device,netdev=net0\
   -netdev user,id=net0,net=192.168.42.0/24,hostfwd=tcp:127.0.0.1:5555-:22\
   -device virtio-serial-device -chardev pty,id=serial3 -device virtconsole,chardev=serial3"

**Comment by xioliu** [comment]
I see, I also need to change cpu_num to 1 in configs\example\baremetal.c

**Reply by danielRep** [reply]
Exactly @xioliu , if you any more issues with bao configuration just take a look also on https://github.com/bao-project/bao-helloworld/ repository. It can provide further help.

**Reply by xioliu** [reply]
Thank you Daniel.

**Comment by xioliu** [comment]
Done
